tools/tools_test.go

```diff
@@ -215,6 +243,29 @@ func TestParser(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:    "qwen two tool calls one with no args",
```

Comment:
We should test the failure case and its expected behavior on how it should fail as well per model

Reply:
Added some more cases